aten/src/ATen/core/jit_type.h

```diff
@@ -391,6 +392,239 @@ struct CAFFE2_API DimensionedTensorType : public TensorType {
   int64_t dim_;
 };
 
+
+template <typename T>
+inline c10::optional<T> merge_primitive (const c10::optional<T>& a, const c10::optional<T>& b)
+{
```

Comment:
nit: are you using the `.clang-format` at the root of pytorch?

Reply:
I usually use `git-clang-format`, so it formats only my changes, but for this PR, I haven't run it yet, since I anticipate that Zach and you will want me to move stuff around.

Reply:
Please run it though :)